invenio_rdm_records/requests/community_inclusion.py

```diff
@@ -10,10 +10,21 @@
 from invenio_i18n import lazy_gettext as _
 from invenio_records_resources.services.uow import RecordCommitOp, RecordIndexOp
 from invenio_requests.customizations import RequestType, actions
+from invenio_requests.errors import CannotExecuteActionError
+
+from invenio_rdm_records.services.errors import InvalidAccessRestrictions
 
 from ..proxies import current_rdm_records_service as service
 
 
+def is_access_restriction_valid(record, community):
+    """Validate that public record cannot be added to restricted community."""
+    is_record_public = record.access.protection.record == "public"
```

Comment:
Indeed, I had the same thoughts after seeing your work for communities. It requires a change everywhere.
For now, I preferred to stick to the existing.

Reply:
I agree, worth opening an issue? for a bright and shiny future when we have time to refactor 🤩 hehe

Reply:
I think this could be easily addressed by @max-moser when he will be working on record access, given that it is quite related?
@max-moser could you refactor the hardcoded strings "public"/"restricted" to use Enum instead, similarly to what done for [communities](https://github.com/inveniosoftware/invenio-communities/blob/9a99e2f540f0ac7126254214d6c7697a55d33a23/invenio_communities/communities/records/systemfields/access.py#L30)?

Reply:
sure thing, could you please open an issue for that up and add it to the sprint board?
i doubt that i'll have time to look at this before leaving for two weeks on thursday :sweat_smile:

Reply:
Added a task in your future issue ;) https://github.com/inveniosoftware/invenio-app-rdm/issues/2180